admin_guide/http_proxies.adoc

```diff
@@ -0,0 +1,159 @@
+= Working with HTTP Proxies
+{product-author}
+{product-version}
+:icons: font
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+:description: This should be a clear and concise description of the topic.
+:keywords: comma, separated, keywords, list
+
+toc::[]
+
+== Overview
+Production environments can deny direct access to the Internet and instead have
+an HTTP or HTTPS proxy available. Configuring OpenShift to use these proxies can
+be as simple as setting standard environment variables in configuration or JSON
+files.
+
+[[configuring-hosts-for-proxies]]
+
+== Configuring Hosts for Proxies
+
+. Add the `*NO_PROXY*`, `*HTTP_PROXY*`, and `*HTTPS_PROXY*` environment variables
+to each host's *_/etc/sysconfig/openshift-master_* or
+*_/etc/sysconfig/openshift-node_* file depending on the type of host:
++
+====
+----
+HTTP_PROXY=http://USERNAME:PASSWORD@10.0.1.1:8080/
+HTTPS_PROXY=https://USERNAME:PASSWORD@10.0.0.1:8080/
+NO_PROXY=master.hostname.example.com
+----
+====
+
+. Restart the master or node as appropriate:
++
+----
+# systemctl restart openshift-master
+# systemctl restart openshift-node
+----
+
+[[proxying-docker-pull]]
+
+== Proxying Docker Pull
+
+OpenShift node hosts need to perform push and pull operations to Docker
+registries. If you have a registry that does not need a proxy for nodes to
+access, include the `*NO_PROXY*` parameter with the registry's hostname, the
+registry service's IP address, and service name. This blacklists that registry's
+address, leaving the external http proxy as the only option.
+
+. Edit the *_/etc/sysconfig/docker_* file and add the variables in shell format:
++
+====
+----
+HTTP_PROXY=http://USERNAME:PASSWORD@10.0.1.1:8080/
+HTTPS_PROXY=https://USERNAME:PASSWORD@10.0.0.1:8080/
+NO_PROXY=master.hostname.example.com,172.30.123.45,docker-registry.default.svc.cluster.local
+----
+====
+
+. Restart the Docker service:
++
+----
+# systemctl restart docker
+----
+
+////
+. If you had already imported ImageStreams before creating the proxy
+configuration, delete, then re-run the process:
++
+----
+# oc delete imagestreams -n openshift --all
+# oc create -f image-streams.json -n openshift
+----
+////
+
+[[s2i-builds]]
+
+== S2I Builds
+
+S2I builds fetch dependencies from various locations. You can
+link:../dev_guide/builds.html#environment-files[use a *_.sti/environment_* file]
+to specify simple shell variables and OpenShift will react accordingly when
+seeing build images.
+
+The following are the supported proxy environment variables with example values:
+
+====
+----
+HTTP_PROXY=http://USERNAME:PASSWORD@10.0.1.1:8080/
+HTTPS_PROXY=https://USERNAME:PASSWORD@10.0.0.1:8080/
+NO_PROXY=master.hostname.example.com
+----
+====
+
+[[configuring-default-templates-for-proxies]]
+
+== Configuring Default Templates for Proxies
+
+The example templates available in OpenShift by default do not include settings
```

Comment:
Maybe link "example templates" here to `install_config/install/first_steps.html#creating-instantapp-templates`?